compiler/rustc_builtin_macros/src/lib.rs

```diff
@@ -66,8 +67,16 @@ pub fn register_builtin_macros(resolver: &mut dyn ResolverExpand) {
     macro register_attr($($name:ident: $f:expr,)*) {
         $(register(sym::$name, SyntaxExtensionKind::LegacyAttr(Box::new($f)));)*
     }
-    macro register_derive($($name:ident: $f:expr,)*) {
-        $(register(sym::$name, SyntaxExtensionKind::LegacyDerive(Box::new(BuiltinDerive($f))));)*
+    macro_rules! register_derive {
+        ($name:ident: $f:expr, $($rest:tt)*) => {
+            register(sym::$name, SyntaxExtensionKind::LegacyDerive(Box::new(BuiltinDerive($f)), smallvec![]));
+            register_derive!($($rest)*);
+        };
+        ($name:ident[$($attr:ident),+]: $f:expr, $($rest:tt)*) => {
```

Comment:
Can we make this syntax a bit more explicit?

```suggestion
        ($name:ident(attrs = [$($attr:ident),+]): $f:expr, $($rest:tt)*) => {
```

so then you write 

```
register! {
  SmartPointer(attrs = [pointee]): ..
}
```

Reply:
Surely. Applied.